data/data/aws/cluster/vpc/vpc-public.tf

```diff
@@ -1,3 +1,7 @@
+locals {
+  has_wavelength_zones = length(compact([for zone_type in var.edge_zones_type : zone_type == "wavelength-zone" ? zone_type : null])) > 0 ? 1 : 0
```

Comment:
The `count` used in the resources consumes that variable and it requires an integer[1], `contains` returns a boolean. Maybe this could work:
~~~
has_wavelength_zones = contains(values(var.edge_zones_type), "wavelength-zone") ? 1 : 0
~~~

Or moving the statement to the resource

[1]
~~~
ERROR Error: Incorrect value type                  
ERROR                                              
ERROR   on vpc/vpc-public.tf line 144, in resource "aws_route_table" "carrier": 
ERROR  144:   count = local.has_wavelength_zones   
ERROR     ├────────────────                        
ERROR     │ local.has_wavelength_zones is a bool 
~~~

Reply:
applyed

Reply:
Right. A pattern we use a lot in our terraform configs is to do:
```
count = local.has_wavelength_zones ? 1 : 0
```

Reply:
> Right. A pattern we use a lot in our terraform configs is to do:
> 
> ```
> count = local.has_wavelength_zones ? 1 : 0
> ```

@r4f4 done, thanks.